networks/rpc/types.go

```diff
@@ -138,6 +140,13 @@ const (
 // - an invalid block number error when the given argument isn't a known strings
 // - an out of range error when the given block number is either too little or too large
 func (bn *BlockNumber) UnmarshalJSON(data []byte) error {
+	var decimalInput uint64
+	err := json.Unmarshal(data, &decimalInput)
+	if err == nil {
+		*bn = BlockNumber(decimalInput)
+		return nil
+	}
```

Comment:
Could you explain why this logic is added though a similar logic exist in the below?

Reply:
BlockNumber(orHash) supports only hex, decimal is not supported.
So if API's input is decimal, param type cannot be BlockNumber. 
I think it was inconsistent. 
This logic is to support decimal(uint64) input in BlockNumber(OrHash).

Reply:
Current
`1` --marshal & BlockNumber.UnmarshlJSON--> fail
`0x1` --marshal & BlockNumber.UnmarshlJSON--> fail
`"1"` --marshal & BlockNumber.UnmarshlJSON-->  fail 
`"0x1"` --marshal & BlockNumber.UnmarshlJSON-->  success 
`"latest"` --marshal & BlockNumber.UnmarshlJSON-->  success 

Your suggestion 
`1` --marshal & BlockNumber.UnmarshlJSON-->  success
`0x1` --marshal & BlockNumber.UnmarshlJSON-->  success
`"1"` --marshal & BlockNumber.UnmarshlJSON-->  fail 
`"0x1"` --marshal & BlockNumber.UnmarshlJSON-->  success 
`"latest"` --marshal & BlockNumber.UnmarshlJSON-->  success 

@hyochan-brown Is it the same with your intention? I think it is little bit confusing.

Reply:
If you intended it, `TestBlockNumberJSONUnmarshal` need to be changed to test `number` type input. Currently, only `string` type inputs have beeing tested.

Reply:
Yes. I intended it. 
Because current BlockNumber cannot unmarshal decimal input, so some API's block number type is integer, not BlockNumber.
I wanted to add consistency in API spec.
I added some test in `TestBlockNumberJSONUnmarshal`.(17-21)
`"n"` is n (number), `'"n"'` is "n" (string) in this test.

Reply:
@hyochan-brown `"n"` is not a number type. It is a string type having a number value. 
```
json.Marshal(1) == []byte{49}
json.Marsahl("1") == []byte{34, 49, 34}
```
Since `TestBlockNumberJSONUnmarshal` doesn't  marshal itsel and use string input, it cannot test number types such int.

Reply:
@hyochan-brown My mistake. You're right. `json.Marshal(1)  == []byte("1")`